ansible_base/lib/jwt_auth/common/auth.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from urllib.parse import urljoin, urlparse
+
+import jwt
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+logger = logging.getLogger("ansible_base.lib.jwt_auth.common.auth")
+
+
+default_mapped_user_fields = [
+    "first_name",
+    "last_name",
+    "email",
+    "is_superuser",
+    "is_system_auditor",
+]
+
+
+class JWTCommonAuth:
+    def __init__(self, user_fields=default_mapped_user_fields) -> None:
+        self.mapped_user_fields = user_fields
+
+    def parse_jwt_token(self, request):
+        logger.debug("Starting JWT Authentication")
+        token = request.headers.get("X-DAB-JW-TOKEN", None)
+        if not token:
+            logger.info("X-DAB-JW-TOKEN header not set for JWT authentication")
+            return None, None
+        logger.debug(f"Received token: {token}")
+
+        jwt_key_setting = None
+        try:
+            jwt_key_setting = settings.ANSIBLE_BASE_JWT_KEY
+        except NameError:
+            logger.info("Failed to get the setting ANSIBLE_BASE_JWT_KEY")
```

Comment:
Well if it requires a setting, does that make it a feature? I would rather not, but maybe this needs to be documented somewhere or maybe have a default in the dynamic settings?

Reply:
No, this is a good callout. We documented that if it requires a feature its an app and not a lib. That being said, I don't think we want to set a default for this because I wouldn't want someone using a "known" JWT key just because they forgot the setting and we catch if its not set and return an actual info message.